drivers/serial/uart_nrfx_uart.c

```diff
@@ -98,6 +102,59 @@ static int baudrate_set(struct device *dev, u32_t baudrate)
 	return 0;
 }
 
+
+#ifdef CONFIG_DEVICE_POWER_MANAGEMENT
+
+static void uart_nrfx_set_power_state(struct device *dev, u32_t power_state)
+{
+	device_power_state = power_state;
+}
+
+static u32_t uart_nrfx_get_power_state(struct device *dev)
+{
+	return device_power_state;
+}
+
+static int uart_nrfx_low_power_device(struct device *dev)
+{
+	if (device_busy_check(dev)) {
+		return -EBUSY;
+	}
+
+	nrf_uart_disable(NRF_UART0);
```

Comment:
I think it may cause, the poll hang. There is a while loop in uart_nrfx_poll_out function
 `while (!nrf_uart_event_check(NRF_UART0, NRF_UART_EVENT_TXDRDY)) {}`
that will not exit if TXRDY event bit is not set. And it will not be set if UART has been disabled.
Am I right @anangl ?

Reply:
Yes. Although the UART is enabled again at resuming from the low power mode, the `STARTTX` task is not triggered at that point. And this will result in hanging in `_poll_out`.

Reply:
How is Rx handled in low power state? interrupt-driven and for RTS/CTS flow control lines?